test-data/unit/check-dataclass-transform.test

```diff
@@ -83,5 +83,88 @@ class A: ...
 @my_dataclass(order=not False)  # E: "order" argument must be True or False.
 class B: ...
 
-[typing fixtures/typing-medium.pyi]
+[typing fixtures/typing-full.pyi]
+[builtins fixtures/dataclasses.pyi]
+
+[case testDataclassTransformDefaultParamsMustBeLiterals]
+# flags: --python-version 3.7
+from typing import dataclass_transform, Type
+
+BOOLEAN_CONSTANT = True
+
+@dataclass_transform(eq_default=BOOLEAN_CONSTANT) # E: eq_default argument must be True or False.
+def foo(cls: Type) -> Type:
+    return cls
+@dataclass_transform(eq_default=(not True)) # E: eq_default argument must be True or False.
+def bar(cls: Type) -> Type:
+    return cls
+
+[typing fixtures/typing-full.pyi]
+[builtins fixtures/dataclasses.pyi]
+
+[case testDataclassTransformUnrecognizedParamsAreErrors]
+# flags: --python-version 3.7
+from typing import dataclass_transform, Type
+
+BOOLEAN_CONSTANT = True
+
+@dataclass_transform(nonexistant=True) # E: unrecognized dataclass_transform parameter 'nonexistant'
+def foo(cls: Type) -> Type:
+    return cls
+
+[typing fixtures/typing-full.pyi]
+[builtins fixtures/dataclasses.pyi]
+
+
+[case testDataclassTransformDefaultParams]
+# flags: --python-version 3.7
+from typing import dataclass_transform, Type, Callable
+
+@dataclass_transform(eq_default=False)
+def no_eq(*, order: bool = False) -> Callable[[Type], Type]:
+    return lambda cls: cls
+@no_eq()
+class Foo: ...
+@no_eq(order=True)
+class Bar: ...  # E: eq must be True if order is True
+
+
+@dataclass_transform(kw_only_default=True)
+def always_use_kw(cls: Type) -> Type:
+    return cls
+@always_use_kw
+class Baz:
+    x: int
+Baz(x=5)
+Baz(5)  # E: Too many positional arguments for "Baz"
+
+@dataclass_transform(order_default=True)
+def ordered(*, eq: bool = True) -> Callable[[Type], Type]:
+    return lambda cls: cls
+@ordered()
+class A:
+    x: int
+A(1) > A(2)
+
+[typing fixtures/typing-full.pyi]
+[builtins fixtures/dataclasses.pyi]
+
+[case testDataclassTransformFieldSpecifiersDefaultsToEmpty]
+# flags: --python-version 3.7
+from dataclasses import field, dataclass
+from typing import dataclass_transform, Type
+
+@dataclass_transform()
+def my_dataclass(cls: Type) -> Type:
+    return cls
+
+@my_dataclass
+class Foo:
+    foo: int = field(kw_only=True)
+
+# Does not cause a type error because `dataclasses.field` is not a recognized field specifier by
+# default
+Foo(5)
+
+[typing fixtures/typing-full.pyi]
```

Comment:
Add a test case where we specify the default and also support overriding it. Example from the PEP:
```py
@typing.dataclass_transform(kw_only_default=True, order_default=True)
def create_model(
    *,
    frozen: bool = False,
    kw_only: bool = True,
) -> Callable[[Type[_T]], Type[_T]]: ...
```

Reply:
added above (`testDataclassTransformDefaultsCanBeOverridden`)